### ISSUE TITLE
[FEATURE] send a notification when a user leaves a chat

### ISSUE THREAD

**Issue by Dun-sin** [Dun-sin/Whisper#307]
### Description

Send a browser notification when one of the user leaves the chat, there's already a pop-up when that happens, but if a user isn't on that page then the user can miss it. 
https://github.com/Dun-sin/Whisper/blob/4b12bf53561e7c2e5f8188c40c1a7d10eba11978/client/src/components/BuddyMatcher.jsx#L91-L106

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

None

**Comment by Rhythm-08** [comment]
Hi, @Dun-sin  I want to solve this issue

**Comment by Dun-sin** [comment]
> Hi, @Dun-sin I want to solve this issue

okay, can you tell me how you would solve it?

**Comment by shreyashnalawade1** [comment]
Hey @Dun-sin,  I have gone through the issues. In my opinion following issues #305 , #306 , #307 , #308 and #309 can be solved if we Create a notification class as a utility that will use the [Notification web Api](https://developer.mozilla.org/en-US/docs/Web/API/notification) and in each place we want to send notifications we can create an object which will accept different options like the message you we want to send, etc. 
I can implement this by the first of October. So can you assign this to me ?

**Comment by Dun-sin** [comment]
> Hey @Dun-sin, I have gone through the issues. In my opinion following issues #305 , #306 , #307 , #308 and #309 can be solved if we Create a notification class as a utility that will use the [Notification web Api](https://developer.mozilla.org/en-US/docs/Web/API/notification) and in each place we want to send notifications we can create an object which will accept different options like the message you we want to send, etc. I can implement this by the first of October. So can you assign this to me ?

no, thanks for wanting to contribute tho, you can check other issues

**Comment by Dun-sin** [comment]
hi @shreyashnalawade1 still interested in this?

**Comment by SiddharthaKR** [comment]
Hi there, I looked into it i wold like to solve this please assign it to me

**Comment by Dun-sin** [comment]
> Hi there, I looked into it i would like to solve this please assign it to me @Dun-sin

how did you look into it?

**Comment by SiddharthaKR** [comment]
@Dun-sin  I looked into the issue and there exist some solutions like
1) Creating a Web Push API for broadcasting the message 
2) Another solution is checking the tab which is active, using this if our tab gets activated our notification will also gets activated
I will try to look which solution matches with the codebase and is robust and lightweight

**Comment by Dun-sin** [comment]
> @Dun-sin I looked into the issue and there exist some solutions like
> 
>     1. Creating a Web Push API for broadcasting the message
> 
>     2. Another solution is checking the tab which is active, using this if our tab gets activated our notification will also gets activated
> 
> 
> I will try to look which solution matches with the codebase and is robust and lightweight

we already have notification implemented, either way i can't assign this to you, will do so if @shreyashnalawade1 doesn't respond in one week

**Comment by dhananjay-Byte** [comment]
hey @Dun-sin can you assign me this issue too?

**Comment by Dun-sin** [comment]
@SiddharthaKR the other dude didn't reply in a week and a week is up, are you still interested in this?

**Comment by SiddharthaKR** [comment]
Hello @Dun-sin i am currently working on different issue, you can assign someone else this issue.
Thank you

**Comment by Nwakaego-Ego** [comment]
Hi Dunsin,

I will love to tag along the person that will be working in this issue if that is ok by them. I am interested in learning how this works

**Comment by Dun-sin** [comment]
> Hi Dunsin,
> 
> I will love to tag along the person that will be working in this issue if that is ok by them. I am interested in learning how this works 

Well that's all up to them

**Comment by dhananjay-Byte** [comment]
hey @Dun-sin if not is picking up this task please assign me this. I have already worked on this and like to learn more about the functionality of the notification API.

**Comment by Dun-sin** [comment]
> hey @Dun-sin if not one is picking up this task please assign me this. I have already worked on this and like to learn more about the functionality of the notification API.

will rather give this to someone else
